### PR TITLE
[JS-to-C++] Test malformed requests

### PR DESCRIPTION
**Pull request by emaxx-google**
Test various edge cases of receiving incorrect PC/SC requests. The only thing to verify is that we don't crash.

This commit contributes to #869.

### CODE REVIEW

**Review by ViktoriiaKovalova — APPROVED** [review]

File: smart_card_connector_app/src/pcsc-api-jstocxxtest.js [hunk 1]
```diff
@@ -141,7 +146,9 @@ function disposeFakeClient(fakeClient) {
  * @return {!Promise}
  */
 async function sleep(timeoutMillis) {
-  return new Promise(resolve => setTimeout(resolve, timeoutMillis));
```
Comment (emaxx-google) [inline smart_card_connector_app/src/pcsc-api-jstocxxtest.js:144]: BTW this change is necessary because the about-to-be-enabled linter would complain on this...

```
  148:33  error  Return values from promise executor functions cannot be read  no-promise-executor-return
```

Luckily we're ready to land #969 now, which would prevent these from sneaking into the tree.
